src/test/java/org/codehaus/mojo/license/api/FreeMarkerHelperTest.java

```diff
@@ -79,10 +76,7 @@ public void testRenderTemplateForThirdPartyFile()
         properties.put( "dependencyMap", licenseMap.toDependencyMap().entrySet() );
 
         String s = helper.renderTemplate( "/org/codehaus/mojo/license/third-party-file.ftl", properties );
-        if ( log.isInfoEnabled() )
-        {
-            log.info( s );
-        }
+        LOG.info( "{}", s );
```

Comment:
If `s` is  a String, why not just `LOG.info(s)` ?

Reply:
Because s could contain `{}`. It's rarely a problem except when you try something like `LOG.warn( s, e )`. If `s` contains `{}`, you get no stack trace anymore.
That's why I always use a format string. In the `buffer.toString()` case, it's more obvious because I save the "turn it into a string" when the logger is disabled.

Reply:
Why would s containing `{}` be a problem? With simple logger, the string is passed as is no matter what it contains: https://github.com/qos-ch/slf4j/blob/master/slf4j-simple/src/main/java/org/slf4j/simple/SimpleLogger.java#L479-L481 and there is no varargs in `Logger.info(String)`. `Logger.info(String)` is the simplest placeholder-less variant.

Reply:
Because in some corner cases, the logging API behaves differently. It works in this case but if you try that with a logger method that takes more arguments, something weird can happen. I prefer to use the same pattern everywhere, so I'm not getting surprised.
My fear is mostly that someone with less knowledge will modify the code later on.